src/core/Rule.ml

```diff
@@ -463,6 +463,21 @@ and step = {
 
 and mode_for_step = [ search_mode | taint_mode ] [@@deriving show]
 
+(*****************************************************************************)
+(* Supply Chain *)
+(*****************************************************************************)
+
+(* You can only do single layer deep OR *)
+type dependency_formula = dependency_pattern list
+
+and dependency_pattern = {
+  (* TODO: actually good types *)
```

Comment:
You can reuse the types in semgrep_output_v1.atd.
It's now accessible from src/core, so you can do

ecosystem: Out.ecosystem

(Out is an alias defined at the top of Rule.ml)